Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.43934&attempt_id=8d44dd9f-65f6-4af8-bc69-3244445499a0)

Attempted: mpd

<details><summary>Partial log (click to expand)</summary><p>

```
                                                          ^
1 error generated.
make[2]: *** [Makefile:558: libnfs_la-socket.lo] Error 1
make[2]: Leaving directory '/private/tmp/nix-build-libnfs-3.0.0.drv-0/source/lib'
make[1]: *** [Makefile:462: all-recursive] Error 1
make[1]: Leaving directory '/private/tmp/nix-build-libnfs-3.0.0.drv-0/source'
make: *** [Makefile:371: all] Error 2
builder for '/nix/store/jal6g9awparfzcj6fwdwg8p4dpmsxrx3-libnfs-3.0.0.drv' failed with exit code 2
cannot build derivation '/nix/store/apnmdplvx8iv0fd9sxw9007nd7qh92iy-mpd-0.20.20.drv': 1 dependencies couldn't be built
error: build of '/nix/store/apnmdplvx8iv0fd9sxw9007nd7qh92iy-mpd-0.20.20.drv' failed
```
</p></details>

--- a/nixos/modules/services/audio/mpd.nix
+++ b/nixos/modules/services/audio/mpd.nix
@@ -55,11 +55,11 @@ in {
       };
 
       musicDirectory = mkOption {
-        type = types.path;
+        type = with types; either path (strMatching "(http|https|nfs|smb)://.+");
         default = "${cfg.dataDir}/music";
         defaultText = ''''${dataDir}/music'';
         description = ''
-          The directory where mpd reads music from.
+          The directory or NFS/SMB network share where mpd reads music from.
         '';
       };
 

--- a/pkgs/servers/mpd/default.nix
+++ b/pkgs/servers/mpd/default.nix
@@ -84,8 +84,8 @@ in stdenv.mkDerivation rec {
     ++ opt clientSupport mpd_clientlib
     ++ opt opusSupport libopus
     ++ opt soundcloudSupport yajl
-    ++ opt nfsSupport libnfs
-    ++ opt smbSupport smbclient;
+    ++ opt (!stdenv.isDarwin && nfsSupport) libnfs
+    ++ opt (!stdenv.isDarwin && smbSupport) smbclient;
 
   nativeBuildInputs = [ autoreconfHook pkgconfig ];
 
@@ -120,8 +120,8 @@ in stdenv.mkDerivation rec {
       (mkFlag clientSupport "libmpdclient")
       (mkFlag opusSupport "opus")
       (mkFlag soundcloudSupport "soundcloud")
-      (mkFlag nfsSupport "libnfs")
-      (mkFlag smbSupport "smbclient")
+      (mkFlag (!stdenv.isDarwin && nfsSupport) "libnfs")
+      (mkFlag (!stdenv.isDarwin && smbSupport) "smbclient")
       "--enable-debug"
       "--with-zeroconf=avahi"
     ]

